app/Http/Controllers/OneChoiceQuestionController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Validator;
+use App\Question;
+use App\QuestionOption;
+use Illuminate\Http\Request;
+use App\Http\Resources\QuestionList as QuestionList;
+use App\Http\Resources\OneChoiceQuestion as OneChoiceQuestion;
+
+class OneChoiceQuestionController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $questions = Question::all();
+
+        return QuestionList::collection($questions);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        Validator::make($request->all(), [
+            'topic_id' => 'required|exists:topics,id',
+            'question' => 'required|string|max:65000',
+            'options' => 'required|array',
+            'options.*.option' => 'required|max:100',
+            'options.*.is_right' => 'required|boolean',
+        ])->validate();
+      
+        $question = Question::create([
+            'topic_id' => $request->topic_id,
+            'type' => Question::TYPE_ONE_CHOICE,
+            'question' => $request->question,
+        ]);
+
+        foreach ($request->options as $option) {
+            $options[] = QuestionOption::make($option);
+        }
+        $question->options()->saveMany($options);
+
+        return OneChoiceQuestion::make($question);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $question = Question::findOrFail($id);
+
+        return OneChoiceQuestion::make($question);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        Validator::make($request->all(), [
+            'topic_id' => 'required|exists:topics,id',
+            'question' => 'required|string|max:65000',
+            'options' => 'required|array',
+            'options.*.option' => 'required|max:100',
+            'options.*.is_right' => 'required|boolean',
+        ])->validate();
+        
+        $question = Question::findOrFail($id);
+        $question->update([
+            'topic_id' => $request->topic_id,
+            'question' => $request->question,
+        ]);
+        $question->options()->delete();
```

Comment:
You may move this line and delete the options directly before adding the new ones.

Reply:
@eduardtovmasyan ?

Reply:
It's a good practice to place actions related to each other in one place. So, for this part, I'll suggest updating the question, then prepare the options and update them. So place this line directly before the line where you're storing the new options.